src/memory.js

```diff
@@ -1,18 +1,41 @@
 class MemoryGame {
   constructor(cards) {
-    this.cards = cards;
-    // add the rest of the class properties here
+    this.cards = cards
+    this.pickedCards = []
+    this.pairsClicked = 0
+    this.pairsGuessed = 0
   }
 
   shuffleCards() {
     // ... write your code here
+    if (!this.cards) {
+      return undefined
+    } else {
+      let chosenIndex = 0
+      for (let i = this.cards.length - 1; i >= 0; i--) {
+
+        chosenIndex = Math.floor(Math.random() * i)
+        let aux = this.cards[chosenIndex]
+        this.cards[chosenIndex] = this.cards[i]
+        this.cards[i] = aux
+
+      }
+    }
+
+    return this.cards
   }
 
-  checkIfPair(card1, card2) {
-    // ... write your code here
+  checkIfPair(nameCard1, nameCard2) { // checkIfPair(pickedCards[0], pickedCards[1])
+    this.pairsClicked++
+    if (nameCard1 === nameCard2) {
+      this.pairsGuessed++
+      return true
+    } else {
+      return false
+    }
   }
 
   checkIfFinished() {
-    // ... write your code here
+    return this.pairsGuessed < 12 ? false : true
```

Comment:
Creo que seria mejor poner this.cards.length / 2 para que valiese para cualquier array de carta